dev-tools/validation/check-design-capability-allocation.py

```
@@ -0,0 +1,934 @@
+    if not match:
+        fail(f"missing section {heading!r}")
+    following = re.search(r"^## ", text[match.end() :], re.MULTILINE)
+    end = match.end() + following.start() if following else len(text)
+    return text[match.end() : end]
+
+
+def table_in_section(text: str, heading: str, required_headers: set[str]) -> tuple[list[str], list[list[str]]]:
+    for headers, rows in markdown_tables(section(text, heading)):
+        if required_headers <= set(headers):
+            return headers, rows
+    fail(f"{heading}: missing table with headers {sorted(required_headers)}")
```

Ruff (0.15.21)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)